src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
+const FULL_HEALTH = 100;
+const BITE_DAMAGE = 50;
+
 class Animal {
-  // write your code here
+  i;
+  static alive = [];
+
+  constructor(name, health = FULL_HEALTH) {
+    this.name = name;
+    this.health = health;
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    if (this.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal !== this);
+    }
```

Comment:
[CODE QUALITY] - The `die` method is performing a filter operation on the `Animal.alive` array directly. Consider moving this operation to a separate method within the `Animal` class to handle removing dead animals from the `Animal.alive` array. This will encapsulate the logic for maintaining the `Animal.alive` array and make the code easier to maintain.